status/projects.py

```diff
@@ -1059,87 +1057,89 @@ class PrioProjectsTableHandler(SafeHandler):
+        for row in itertools.chain.from_iterable(view_calls):
+            proj_id = row.key[1]
+            proj_val = row.value
+            for date_type, date in proj_val['summary_dates'].items():
+                proj_val[date_type] = date
+            if row.key[0] == 'ongoing':
+                for k, v in proj_val['project_summary'].items():
+                    proj_val[k] = v
+
+            is_fin_lib = False
+            if 'by user' in proj_val.get('library_construction_method', '-').lower():
+                is_fin_lib = True
+
+            for key, value in def_dates_rec_ctrl.items():
+                start_date = value[0]
+                end_date = value[1]
+                date_val = self._calculate_days_in_status(proj_val.get(start_date),
+                                                               proj_val.get(end_date))
+                projects[proj_id] = { key: date_val }
+
+            if row.key[0] == 'ongoing':
+                for key, value in def_dates_ongoing.items():
+                    if key == 'days_seq_start':
+                        if is_fin_lib:
+                            start_date = value[0][1]
+                        else:
+                            start_date = value[0][0]
+                    else:
+                        start_date = value[0]
+                        end_date = value[1]
+                    if key in ['days_prep', 'days_prep_start'] and is_fin_lib:
+                        date_val = 0
+                    else:
+                        date_val = self._calculate_days_in_status(proj_val.get(start_date),
+                                                                       proj_val.get(end_date))
+
+                    projects[proj_id][key] = date_val
+
 
         #Delete statuses with 0 days
-        days_dict = copy.deepcopy(statuses_dict)
-        for k,v in statuses_dict.items():
+        for k,v in projects.items():
             min_days = 0
-            for k2,v2 in v.items():
-                if v2 <= min_days:
-                    del days_dict[k][k2]
- 
+            for k2 in list(v):
+                if v[k2] <= min_days:
+                    del projects[k][k2]
+
         #Get list of projects with status and days containing only last status
-        table_data = [(k,k2,v2) for k,v in days_dict.items() for k2,v2 in v.items() if k2 in list(v.items())[-1]]
+        t_data = [(k,k2,v2) for k,v in projects.items() for k2,v2 in v.items() if k2 in list(v.items())[-1]]
         #Sort projects on number of days
-        table_data.sort(key=lambda x:x[2], reverse=True)
-        #Get top 15 projects
-        self.write(json.dumps(table_data[:15]))
+        t_data.sort(key=lambda x:(x[1], x[2]), reverse=True)
+        table_data = []
+        #Get top 15 projects in each status
+        for key in list(def_dates_rec_ctrl.keys())+list(def_dates_ongoing.keys()):
+            table_data = table_data + [item for item in t_data if item[1]==key][:15]
```

Comment:
Yes, I was thinking we wanted the projects that have been waiting the longest, regardless of the status. 
But it would look a lot better if all cells of the table would be populated, and since the number of days is included and the projects are sorted, I guess we might as well add the top projects of each status...

Do you think that 15 is a good number?